backend/src/main/kotlin/app/ehrenamtskarte/backend/application/database/repos/ApplicationRepository.kt

```diff
@@ -109,6 +124,18 @@ object ApplicationRepository {
         }
     }
 
+    fun withdrawApplication(accessKey: String): Boolean {
+        return transaction {
+            val application = ApplicationEntity.find { Applications.accessKey eq accessKey }.single()
+            if (application != null) {
```

Comment:
I believe `.single` should throw if there is no element, and hence checking for null should not be necessary?

Reply:
thats true. I think i should check if `withdrawalDate == null` before setting the date and else throw an exception that its already set

Reply:
removed check